### PR TITLE
Vagga config

### PR DESCRIPTION
**Pull request by anxolerd**
# Vagga config
## Summary

Keep working on vagga config
## Changelog
- Added new container for running tests
- Added command for running nosetests 
- Better run command description
- Do not use pidfile when running app in development mode
- Decouple vagga from makefile. In fututre it would be great to get rid of that scary makefile.

### CODE REVIEW

File: vagga.yaml [hunk 1]
```diff
@@ -22,22 +17,46 @@ containers:
...
+      NOSE_TESTCONFIG_AUTOLOAD_YAML: "config/test/app.yml"
 
 commands:
   run: !Supervise
-  description: Run app - testing status
-  children:
-    env: !Command
-    container: gdg_dev
-    run: |
-      make run-dev
```
Comment (webknjaz) [inline vagga.yaml:43]: Bad idea. You'd have to support separate documentation/process explaining how to run everything on Mac environments.

File: vagga.yaml [hunk 2]
```diff
@@ -22,22 +17,46 @@ containers:
...
+      - !BuildDeps
+        - git
+        - mercurial
+      - !Depends requirements/common.txt
+      - !Depends requirements/dev.txt
+      - !Py3Requirements requirements/test.txt
+      - !Py3Requirements requirements/test-env.txt
+    environ:
+      BLUEBERRYPY_CONFIG: "{}"
+      NOSE_TESTCONFIG_AUTOLOAD_YAML: "config/test/app.yml"
```
Comment (webknjaz) [inline vagga.yaml:38]: You don't have to wrap a normal string with quotes

File: vagga.yaml [hunk 3]
```diff
@@ -22,22 +17,46 @@ containers:
...
-      - !EnsureDir /var/tmp/run
     environ:
       APP_ROOT: /work  # Make sure this is not overriden by .exports
-    volumes:
-      /var/tmp/run: !Tmpfs
-        size: 5Mi
-        mode: 0o1777
+
+  gdg-test:
+    setup:
```
Comment (webknjaz) [inline vagga.yaml:27]: I'd add `dependencies: true` here as well

File: Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby ts=2 sw=2 tw=2 et :
+
+
+$script = <<SCRIPT
+VAGGA_CONF_FILE='/home/vagrant/.vagga/settings.yaml'
+if ! grep -q "ubuntu-mirror: " $VAGGA_CONF_FILE
```
Comment (webknjaz) [inline Vagrantfile:7]: Why do you do this?

File: Vagrantfile [hunk 5]
```diff
@@ -0,0 +1,35 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby ts=2 sw=2 tw=2 et :
+
+
+$script = <<SCRIPT
```
Comment (webknjaz) [inline Vagrantfile:5]: ?

File: Vagrantfile (hunk 4 above)
Comment (anxolerd) [inline Vagrantfile:7]: https://github.com/tailhook/vagga/issues/263

Comment (webknjaz) [inline Vagrantfile:7]: Please state minimum explanation needed as a comment here with links provided.
The reason should be explicitly clear and obvious from the text.

**Pull request closed without merging**